lib/kitchen-sink/map.ex

```diff
@@ -138,4 +138,40 @@ defmodule KitchenSink.Map do
   def rename_key(_,_,_) do
     %{}
   end
+
+  def new_nested_map(key_list, value) do
```

Comment:
new_nested_map is close to the Map.new name.
maybe KMap.new_nested. or KMap.nested?
i don't like the "make" prefix as i don't know of other examples using this prefix